### PR TITLE
resolve issues #19, #20 and #22

### PR DESCRIPTION
**Pull request by fschloesser**
I guess I didn't do the branching right, but I know that now and it will do that better in the future.

### CODE REVIEW

File: test/EvaluationTest.py [hunk 1]
```diff
@@ -24,8 +27,87 @@
...
+testevalstr_col = ['\t<Column name="Time" origcolname="SolvingTime" alternative="100" reduction="mean">\n{}\t</Column>\n',
+                   '\t<Column origcolname="ProblemName" reduction="strConcat">\n{}\t</Column>\n',
+                   '\t<Column name="Indic" constant="0" alternative="1" reduction="max">\n\
+\t\t<Filter expression1="SolvingTime" expression2="10" operator="ge" />\n{}\n\t</Column>\n']
+testevalstr_agg = ['\t\t<Aggregation aggregation="mean" />\n',
+                   '\t\t<Aggregation aggregation="strConcat" />\n',
+                   '\t\t<Aggregation aggregation="strConcat" />\n']
+testevalstr_fg = ['\t<FilterGroup active="True" filtertype="intersection" name="all"/>\n',
+                  '\t<FilterGroup name="hard">\n\t\t<Filter anytestrun="any" expression1="Time" expression2="100" operator="ge"/>\n\t</FilterGroup>\n']
+testevalstr_foot = '</Evaluation>'
```
Comment (fschloesser) [inline test/EvaluationTest.py]: Simply did not think about that. Is changed now and using constructors.

**Comment by fschloesser** [comment]
I updated the code, can you please review and merge it?

File: ipet/evaluation/IPETEvalTable.py [hunk 2]
```diff
@@ -1145,6 +1238,11 @@ def evaluate(self, exp : Experiment):
         data = exp.getJoinedData()
         logging.debug("Result of getJoinedData:\n{}\n".format(data))
 
+        if self.autoIndex:
```
Comment (GregorCH) [inline ipet/evaluation/IPETEvalTable.py]: I think we should hide this `if` inside generateIndex.

File: ipet/evaluation/IPETEvalTable.py [hunk 3]
```diff
@@ -1123,6 +1148,74 @@ def getAggregatedData(self):
...
+            second = [sorted_indices[0]]
+            # check if a second columnindex can be helpful
+            if len(sorted_indices) > 1 and (height / first[1]) / second[0][1] > 1:
+                second.append(sorted_indices[1])
+
+        # set everything
+        self.indexsplit = 1
+        self.generateDefaultGroup(data, [i[0] for i in second])
+        self.set_index(" ".join([i[0] for i in [first] + second]))
+        logging.debug("Automatically set index to ({}, {})".format(self.getRowIndex(), self.getColIndex()))
```
Comment (GregorCH) [inline ipet/evaluation/IPETEvalTable.py]: This is rather an info message

File: ipet/evaluation/IPETEvalTable.py [hunk 4]
```diff
@@ -1145,6 +1238,11 @@ def evaluate(self, exp : Experiment):
         data = exp.getJoinedData()
         logging.debug("Result of getJoinedData:\n{}\n".format(data))
 
+        if self.autoIndex:
+            self.generateIndex(data)
+        elif self.orig_defaultgroup == "":
+            self.generateDefaultGroup(data, list(self.getColIndex()))
```
Comment (GregorCH) [inline ipet/evaluation/IPETEvalTable.py]: I think an info message that a default group was computed is helpful.

**Comment by GregorCH** [comment]
Nice work, I made a couple of comments, then we are good to merge.

**Comment by fschloesser** [comment]
good, can i merge it like this?

**Comment by GregorCH** [comment]
No, first you have to change the description and mention the issues that this fixes :)

File: ipet/evaluation/IPETEvalTable.py [hunk 5]
```diff
@@ -1190,32 +1191,36 @@ def generateIndex(self, data):
...
-        
-        # find the indices that are represented in the data with their numbers of unique values
-        present_indices = [[key, data[key].nunique()] for key in possible_indices if key in data.columns]
-        # take the index with the max value of the previous as rowindex
-        first = max(present_indices, key = lambda y: y[1])
-
-        processed_indices = [[key, count] for [key, count] in present_indices if count > lowerbound and key != first[0]]
-        sorted_indices = sorted(processed_indices, key = lambda y: y[1])
+        # do this only if the user requested an automatic index
+        if self.autoIndex:
```
Comment (GregorCH) [inline ipet/evaluation/IPETEvalTable.py]: How about `if not self.autoIndex: return`

File: ipet/evaluation/IPETEvalTable.py [hunk 6]
```diff
@@ -1173,6 +1173,7 @@ def generateDefaultGroup(self, data, colindex):
 
         # take the tuple with the highest occurrence
         self.orig_defaultgroup = ":".join(max_key)
+        logging.info("Generated defaultgroup to {}".format(self.orig_defaultgroup))
```
Comment (GregorCH) [inline ipet/evaluation/IPETEvalTable.py]: Revise English. I would say "Using {} as default group"

Comment (fschloesser) [inline ipet/evaluation/IPETEvalTable.py]: ok

File: ipet/evaluation/IPETEvalTable.py (hunk 5 above)
Comment (fschloesser) [inline ipet/evaluation/IPETEvalTable.py]: ok

**Comment by fschloesser** [comment]
well too bad :/

**Pull request merged**